### DISCUSSION TITLE
How to restore the graffiti area of the original image when restoring a pre edited image

### DISCUSSION THREAD

**Discussion by zhenzi0322** [fabricjs/fabric.js#9994] [Ideas]
Similar effects are as follows

![2024-07-13_10-27-19](https://github.com/user-attachments/assets/7fcf51a4-ca55-4426-b08e-5f4809970cfe)

When I am doodling, there will be a re rendering. How is this usage done

**Comment by asturur** [comment]
Is completely unclear what are you doing.
Since is 2024 i suspect you are talking about a background removal kind of application.
 
You have a picture to which a mask is applied and removes the background.
What you can do is to  take your path, send it to the plane of the image, draw it with a globalCompositeOperation that adds opaque pixels back to the mask.

The details of this is entirely dependent on your application setup
